flake.nix

```diff
@@ -1,126 +1,347 @@
+      flake = false;
+    };
+    patch-osg = {
+      url = "https://github.com/NixOS/nixpkgs/pull/384430.patch";
+      flake = false;
+    };
+    patch-piqp = {
+      url = "https://github.com/NixOS/nixpkgs/pull/374657.patch";
+      flake = false;
+    };
+    src-agimus-msgs = {
+      url = "github:agimus-project/agimus_msgs";
+      flake = false;
+    };
+    src-colmpc = {
+      url = "github:agimus-project/colmpc";
+      flake = false;
+    };
+    src-example-parallel-robots = {
+      url = "github:gepetto/example-parallel-robots";
+      flake = false;
+    };
+    src-franka-description = {
+      url = "github:agimus-project/franka_description";
       flake = false;
     };
-
     # gepetto-viewer has a fix to understand AMENT_PREFIX_PATH in #239/devel
-    gepetto-viewer = {
+    src-gepetto-viewer = {
       url = "github:Gepetto/gepetto-viewer/devel";
-      inputs.nixpkgs.follows = "nixpkgs";
+      flake = false;
     };
+    src-linear-feedback-controller = {
+      url = "github:loco-3d/linear-feedback-controller";
+      flake = false;
+    };
+    src-linear-feedback-controller-msgs = {
+      url = "github:loco-3d/linear-feedback-controller-msgs";
+      flake = false;
+    };
+    src-toolbox-parallel-robots = {
+      url = "github:gepetto/toolbox-parallel-robots";
+      flake = false;
+    };
+
+    # keep-sorted end
   };
   outputs =
-    { nixpkgs, self, ... }@inputs:
-    inputs.nix-ros-overlay.inputs.flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        pkgs = import ./patched-nixpkgs.nix {
-          inherit nixpkgs system;
-          overlays = [
-            inputs.nix-ros-overlay.overlays.default
-            (_super: prev: {
-              gepetto-viewer = prev.gepetto-viewer.overrideAttrs {
-                inherit (inputs.gepetto-viewer.packages.${system}.gepetto-viewer) src;
-              };
-            })
-          ];
-          patches = [
-            inputs.patch-brax
-            inputs.patch-hpp
+    inputs:
+    let
+      pkgsForPatching = inputs.nixpkgs.legacyPackages.x86_64-linux;
+      patches = [
+        # sort this by patch application order
+        # not alphabetically
+        inputs.patch-crocoddyl
+        inputs.patch-aligator
+        inputs.patch-brax
+        inputs.patch-hpp
+        inputs.patch-mim-solvers
+        inputs.patch-osg
+        inputs.patch-piqp
+      ];
+      patchedNixpkgs = (
+        pkgsForPatching.applyPatches {
+          inherit patches;
+          name = "patched nixpkgs";
+          src = inputs.nixpkgs;
+        }
+      );
+      overlay = (
+        final: prev:
+        {
+          inherit (inputs)
+            # keep-sorted start
+            src-agimus-msgs
+            src-colmpc
+            src-linear-feedback-controller
+            src-linear-feedback-controller-msgs
+            # keep-sorted end
+            ;
+          # keep-sorted start block=yes
```

Comment:
Maybe just after:
```nix
      overlay = (
        final: prev:
        {
```
??

Reply:
sorry, what are you talking about here ?

Reply:
would need to play more with keep-sorted yes